Jenkinsfile

```diff
@@ -0,0 +1,335 @@
+                  doGenerateSubmoduleConfigurations: false, 
+                  extensions: [],
+                  submoduleCfg: [],
+                  userRemoteConfigs: [[credentialsId: "${JOB_GIT_CREDENTIALS}", 
+                  url: "${GIT_URL}"]]])
+              }
+              env.JOB_GIT_COMMIT = getGitCommitSha()
+              env.JOB_GIT_URL = "${GIT_URL}"
+              
+              github_enterprise_constructor()
+              
+              on_change{
+                echo "build cause a change (commit or PR)"
+              }
+              
+              on_commit {
+                echo "build cause a commit"
+              }
+              
+              on_merge {
+                echo "build cause a merge"
+              }
+              
+              on_pull_request {
+                echo "build cause PR"
+              }
+              gitCreateTag()
+            }
+          }
+          stash allowEmpty: true, name: 'source'
+        }
+      }
+    }
+    
+    /**
+    Build on a linux environment.
+    */
+    stage('build') { 
+      agent { label 'linux' }
+      environment {
+        PATH = "${env.PATH}:${env.HUDSON_HOME}/go/bin/:${env.WORKSPACE}/bin"
+        GOPATH = "${env.WORKSPACE}"
+      }
+      
+      when { 
+        beforeAgent true
+        environment name: 'linux_ci', value: 'true' 
+      }
+      steps {
+        withEnvWrapper() {
+          unstash 'source'
+          dir("${BASE_DIR}"){    
+            sh """#!/bin/bash
+            make install check
+            """
+          }
+        }
+      }
+    }
+    stage('Parallel stages') {
+      failFast true
+      parallel {
+        stage('test') {
+          agent { label 'linux' }
+          environment {
+            PATH = "${env.PATH}:${env.HUDSON_HOME}/go/bin/:${env.WORKSPACE}/bin"
+            GOPATH = "${env.WORKSPACE}"
+          }
+          
+          when { 
+            beforeAgent true
+            environment name: 'test_ci', value: 'true' 
+          }
+          steps {
+            withEnvWrapper() {
+              unstash 'source'
+              dir("${BASE_DIR}"){    
+                sh """#!/bin/bash
+                ./scripts/jenkins/test.sh
+                """
+                codecov('apm-agent-go')
+              }
+            }
+          }
+          post { 
+            always {
+              coverageReport("${BASE_DIR}/build")
+              junit(allowEmptyResults: true, 
+                keepLongStdio: true, 
+                testResults: "${BASE_DIR}/build/junit-*.xml")
+            }
+          }
+        }
+        stage('Benchmarks') {
+          agent { label 'linux' }
+          environment {
+            PATH = "${env.PATH}:${env.HUDSON_HOME}/go/bin/:${env.WORKSPACE}/bin"
+            GOPATH = "${env.WORKSPACE}"
+          }
+          
+          when { 
+            beforeAgent true
+            allOf { 
+              //branch 'master';
```

Comment:
```suggestion
```
?

Reply:
this step will executed only on the master branch, for the moment is comments until we test all the pipeline.